### PR TITLE
[Link][a11y] Implemented accesibilityLabel prop

### PR DESCRIPTION
**Pull request by ErickTamayo**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-ux/issues/458 

### WHAT is this pull request doing?

Adds the `accessibilityLabel` prop to the `Link` component

## 

### How to 🎩

- Add the `accessibilityLabel` prop to the `Link` component and observe the `aria-label` prop is added to the component

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}

```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [X] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [X] Updated the component's `README.md` with documentation changes
* [X] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to format new entries. 💜
 
+- Adds `accessibilityLabel` to `Link` component ([#3691](https://github.com/Shopify/polaris-react/pull/3691))
```
Comment (kyledurand) [inline UNRELEASED.md:5]:
```suggestion
- Added `accessibilityLabel` to `Link` component ([#3691](https://github.com/Shopify/polaris-react/pull/3691))
```

We opt for past tense changelog additions

**Comment by ErickTamayo** [comment]
@BPScott This is the new PR with the recommendations of https://github.com/Shopify/polaris-react/pull/3668

File: src/components/Link/tests/Link.test.tsx [hunk 2]
```diff
@@ -123,4 +123,30 @@ describe('<Link />', () => {
...
+
+      expect(link).toContainReactComponent('button', {
+        'aria-label': mockAccessibilityLabel,
+      });
+    });
+
+    it('passes prop', () => {
+      const mockAccessibilityLabel = 'mock accessibility label';
+
+      const link = mountWithApp<LinkProps>(
```
Comment (kyledurand) [inline src/components/Link/tests/Link.test.tsx:142]: Why is this needed? I thought TS would interpolate the props just fine here

File: src/components/Link/tests/Link.test.tsx [hunk 3]
```diff
@@ -123,4 +123,30 @@ describe('<Link />', () => {
...
+        <Link
+          url="https://shopify.com"
+          accessibilityLabel={mockAccessibilityLabel}
+        />,
+      );
+
+      expect(link.prop('accessibilityLabel')).toBe(mockAccessibilityLabel);
```
Comment (kyledurand) [inline src/components/Link/tests/Link.test.tsx:149]:
```suggestion
      expect(link).toContainReactComponent('a', {
        'aria-label': mockAccessibilityLabel,
      });
```
Should work here

**Review by kyledurand — APPROVED** [review]
Looks good, just some super small comments

File: src/components/Link/tests/Link.test.tsx (hunk 2 above)
Comment (ErickTamayo) [reply src/components/Link/tests/Link.test.tsx:142]: For some reason is not getting interpolated correctly: 
image

File: src/components/Link/tests/Link.test.tsx [hunk 4]
```diff
@@ -123,4 +123,32 @@ describe('<Link />', () => {
...
+      const mockAccessibilityLabel = 'mock accessibility label';
+
+      const link = mountWithApp<LinkProps>(
+        <Link
+          url="https://shopify.com"
+          accessibilityLabel={mockAccessibilityLabel}
+        />,
+      );
+
+      expect(link).toContainReactComponent('a', {
```
Comment (kyledurand) [inline src/components/Link/tests/Link.test.tsx:149]: My bad. I just left an edit saying I think this needs to be calling `UnstyledLink` vs `'a'`. Just did a double check of the component

Comment (ErickTamayo) [reply src/components/Link/tests/Link.test.tsx:149]: Alright I'll correct it

File: src/components/Link/tests/Link.test.tsx [hunk 5]
```diff
@@ -123,4 +123,32 @@ describe('<Link />', () => {
...
+      const link = mountWithApp(
+        <Link accessibilityLabel={mockAccessibilityLabel} />,
+      );
+
+      expect(link).toContainReactComponent('button', {
+        'aria-label': mockAccessibilityLabel,
+      });
+    });
+
+    it('passes prop', () => {
```
Comment (kyledurand) [inline src/components/Link/tests/Link.test.tsx:139]:
```suggestion
    it('passes the accessibilityLabel to UnstyledLink when url is present', () => {
```
